templates/repo/code/upstream_diverging_info.tmpl

```diff
@@ -0,0 +1,12 @@
+{{if and .UpstreamDivergingInfo .UpstreamDivergingInfo.CommitsBehind}}
+<div class="ui message flex-text-block">
+	<div class="tw-flex-1">
+		{{$upstreamLink := printf "%s/src/branch/%s" .Repository.BaseRepo.Link .BranchName}}
+		{{$upstreamHtml := HTMLFormat `<a href="%s">%s:%s</a>` $upstreamLink .Repository.BaseRepo.FullName .BranchName}}
+		{{ctx.Locale.TrN .UpstreamDivergingInfo.CommitsBehind "repo.pulls.upstream_diverging_prompt_behind_1" "repo.pulls.upstream_diverging_prompt_behind_n" .UpstreamDivergingInfo.CommitsBehind $upstreamHtml}}
+	</div>
+	<button class="ui compact green button tw-m-0 link-action" data-url="{{.Repository.Link}}/branches/merge-upstream?branch={{.BranchName}}">
+		{{ctx.Locale.Tr "repo.pulls.upstream_diverging_merge"}}
+	</button>
```

Comment:
Need to hide the button for logged-out users?

Reply:
Not just logged out users, also user who only have read permission on the repo

Reply:
Oh yes, forgot to hide the button on UI, done in fc832132c992e54b33490b987fba2e237ec31047